cli/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -711,9 +713,39 @@ pub fn path_hash(p: &impl AsRef<Path>) -> String {
     chars.to_string()
 }
 
+/// Return a path to the services socket given a unique identifier
+///
+/// Socket paths cannot exceed 104 characters on macOS
+/// - TMPDIR will often have a long path, e.g.
+///   /var/folders/8q/spckhr654cv4xrcv0fxsrlvc0000gn/T/nix-shell.vfDA8u
+/// - /var/run is not writeable
+/// So we use /tmp
```

Comment:
(non-blocking observations) `/tmp` symlinks to `/private/tmp` but using the former should be fine. There are some notes in `periodic.conf(5)` about a mechanism for automatically cleaning temp directories and a long running activation could fall foul of not accessing the socket in N days but these don't appear to be enabled automatically on my (Sonoma) machine.

> The following variables are used by the standard scripts that reside in /etc/periodic/daily:
> 
>          daily_clean_tmps_enable
>              (bool) Set to “YES” if you want to clear temporary directories daily.
> 
>          daily_clean_tmps_dirs
>              (str) Set to the list of directories to clear if daily_clean_tmps_enable is set to “YES”.
> 
>          daily_clean_tmps_days
>              (num) When daily_clean_tmps_enable is set, this must also be set to the number of days old that a file's access and modification times must be before it is deleted.
> 
>          daily_clean_tmps_ignore
>              (str) Set to the list of files that should not be deleted when daily_clean_tmps_enable is set to “YES”.  Wild card characters are permitted.
> 
>          daily_clean_tmps_verbose
>              (bool) Set to “YES” if you want the removed files to be reported in your daily output.
> 
>          daily_local
>              (str) Set to a list of extra scripts that should be run after all other daily scripts.  All scripts must be absolute path names.

Reply:
@limeytexan I think you've run into issues with `/tmp` getting cleaned up, was that on Linux?

Reply:
Would `~/.cache/flox/run/services.path_hash.sock` be better? That would typically be 46 + len(username)

Reply:
I have been wavering on suggesting it. It's not as "standard" but it does feel like it would be safer and also consistent across all platforms.

Reply:
Changed it to fall back to `~/.cache/flox`